src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -78,13 +78,15 @@ internal static class WinHttpResponseParser
                     if (contentEncodingLength > 0)
                     {
                         if (CharArrayHelpers.EqualsOrdinalAsciiIgnoreCase(
-                            EncodingNameGzip, buffer, contentEncodingStartIndex, contentEncodingLength))
+                            EncodingNameGzip, buffer, contentEncodingStartIndex, contentEncodingLength) &&
+                            (manuallyProcessedDecompressionMethods & DecompressionMethods.GZip) == DecompressionMethods.GZip)
```

Comment:
I'd suggest swapping the order in these checks, so that we only do the more expensive string comparison if the less expensive enum comparison passes.

Reply:
Good idea.